[2주차] 서나영/[feat] 로그인, 회원가입 UI 구현

nabbang6 opened the pull request:
## 1. 무슨 이유로 코드를 변경했나요?

- 로그인 Modal 컴포넌트 구현
- 회원가입 페이지 UI 구현
- Header, Sidebar 컴포넌트 구현
- 라우터 설정 

<br> 

## 2. 어떤 위험이나 장애를 발견했나요?

현재 Header에서 GITLOG 로고가 보이지 않는 오류가 있습니다.. 
문제 원인 파악하면 바로 수정할 예정입니다 ㅜ__ㅜ!

<br>

## 3. 관련 스크린샷을 첨부해주세요. 

- **LoginModal** 
![image](https://github.com/user-attachments/assets/67f20b16-4632-4f78-8a8d-df8ff24baeec)

- **Header 컴포넌트** 
![image](https://github.com/user-attachments/assets/7bdc09d5-5b49-48cf-9c18-9e70e323b7a9)

- **Sidebar 컴포넌트** 
![image](https://github.com/user-attachments/assets/8e819c22-225e-42d6-9056-516ac428913e)

- **회원가입 페이지** 
![image](https://github.com/user-attachments/assets/fe0641b2-3240-48e3-8d18-56c0b1e0f937)

<br>

## 4. 완료 사항

1. global style 적용
2. Header 컴포넌트 작성
3. Sidebar 컴포넌트 작성
4. LoginModal 컴포넌트 작성
5. 회원가입 페이지 ui 구현
6. 라우터 설정 

close #17

nabbang6 committed:
feat: svg 파일 추가 및 assets/icons로 이동, index.tsx 추가

nabbang6 committed:
refactor: 기존 css 제거 & global.ts로 적용

nabbang6 committed:
refactor: 공통 컴포넌트들을 src/components/ui로 이동

nabbang6 committed:
feat: SVG 아이콘 export

nabbang6 committed:
chore: tsconfig.json 타입 경로 설정 추가

nabbang6 committed:
chore: ESLint에 Prettier 설정 추가

nabbang6 committed:
feat: 필요한 SVG 추가

nabbang6 committed:
remove: 불필요한 style 코드 삭제

nabbang6 committed:
feat: ttf파일을 모듈로 쓸 수 있게 선언

nabbang6 committed:
fix: 아이콘 불러오는 방식 수정

nabbang6 committed:
feat: Header 컴포넌트 구현

nabbang6 committed:
feat: Sidebar 컴포넌트 구현

nabbang6 committed:
feat: Login Modal 컴포넌트 구현

nabbang6 committed:
fix: styled-components 프롭 명칭 변경 (maxWidth -> $ maxWidth)

nabbang6 committed:
feat: 회원가입 페이지 Header UI 구현

nabbang6 committed:
feat: 회원가입 페이지 UI 구현

nabbang6 committed:
feat: 전체 회원가입 페이지 UI 구현

nabbang6 committed:
refactor: Routes 구조 변경 및 회원가입 페이지 연결
- 기존 main.tsx에서 Routes 설정을 분리하다 App.tsx에서 관리하도록 변경
- 공통 컴포넌트 예시 삭제

One-HyeWon commented on 서나영-Blog/src/assets/icons/GITLOG.svg:
GITLOG 아이콘을 화이트로 가져오셔서 헤더에서 안 보이는 것 같아요!!
svgr 라이브러리를 사용하고 계시니 svg 파일에서 fill='current'로 주시고

GITLOG svg 컴포넌트 호출 하실 때
<GITLOG fill='#~~~' />으로, 헤더에서 사용 되고 있는 로고의 색상을 지정 해 주시면 보일 것 같아요!!
svgr 라이브러리는 이런 current 값 할당을 통해 동적으로 속성 할당이 가능합니다!

nabbang6 replied:
헉 svg에 요런 방법으로 속성 할당 가능한지는 처음 알았네요... 
수정했더니 이제 잘 보입니다!!👍👍🥹 감사합니당

One-HyeWon commented on 서나영-Blog/src/components/auth/signup/SignupForm.tsx:
```diff
@@ -0,0 +1,113 @@
+      </ButtonWrapper>
+    </FormContainer>
+  );
+};
```
이부분 코드 반복이 많아서, 상태 관리를 객체로 하는 것은 어떤가 제안 해 봅니다!!

```
 // 폼 필드를 객체로 관리
  const [formData, setFormData] = useState({
    email: '',
    password: '',
    confirmPassword: '',
    name: '',
    birthDate: '',
    nickname: '',
    bio: ''
  });

  // 모든 입력 필드의 변경을 처리하는 단일 핸들러
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData({
      ...formData,
      [name]: value
    });
  };

  // 회원가입 제출 핸들러
  const handleSubmit = () => {
    console.log('회원가입 데이터:', formData);
    // 여기에 유효성 검사 및 API 호출 로직 추가
  };

  // 입력 필드 설정을 배열로 정의
  const inputFields = [
    { name: 'email', label: '이메일', type: 'email', placeholder: '이메일' },
    { name: 'password', label: '비밀번호', type: 'password', placeholder: '비밀번호' },
    { name: 'confirmPassword', label: '비밀번호 확인', type: 'password', placeholder: '비밀번호 확인' },
    { name: 'name', label: '이름', type: 'text', placeholder: '이름' },
    { name: 'birthDate', label: '생년월일', type: 'date', placeholder: 'YYYY-MM-DD' },
    { name: 'nickname', label: '닉네임', type: 'text', placeholder: '닉네임' },
    { name: 'bio', label: '한 줄 소개', type: 'text', placeholder: '한 줄 소개' }
  ];

  return (
    <FormContainer>
      <ProfileUpload />
      
      {/* 입력 필드 매핑 */}
      {inputFields.map((field) => (
        <SignupInput
          key={field.name}
          name={field.name}
          label={field.label}
          type={field.type}
          value={formData[field.name]}
          onChange={handleChange}
          placeholder={field.placeholder}
        />
      ))}
      
      <ButtonWrapper>
        <Button
          width='100%'
          onClick={handleSubmit}
          style={{
            color: '#00A1FF',
            borderRadius: '25px',
            backgroundColor: '#FFF',
            border: '1px solid #00A1FF',
          }}
        >
          회원가입 완료
        </Button>
      </ButtonWrapper>
    </FormContainer>
  );
```

입력 필드 설정 객체 정의가 지저분해 보일 수도 있는데, 추후에 입력 필드 구성 요소에 변경 사항이 생길 수도 있다는 가정 하에 객체 형식으로 관리 하는 것이 편할 것 같아요!

nabbang6 replied:
아하 반영했습니다!! 객체로 한 번에 관리할 수 있어서 확실히 더 좋은 것 같아요,,,👍

One-HyeWon commented on 서나영-Blog/src/components/layout/header/ChatandMore.tsx:
```diff
@@ -0,0 +1,31 @@
+  gap: 10px;
+`;
+
+const ChatandMore = () => (
```
와 저도 헤더 컴포넌트 이렇게 구현해서 신기하네요
이름도 똑같아요

One-HyeWon commented on 서나영-Blog/src/components/modal/LoginModal.tsx:
```diff
@@ -0,0 +1,271 @@
+          </Button>
+          <SnsContent>
+            <Divider />
+            <span>SNS</span>
```
이 텍스트는 스타일을 따로 주지 않아도 되던가요?

nabbang6 replied:
앗 `SnsContent` 스타일 정의할 때, 해당 컴포넌트 내부에 있는 `span` 태그에 따로 스타일 적용하도록 구현해뒀습니다! 좀 더 직관적으로 `span` 태그 자체에 직접 스타일 지정해주는 편이 좋을까요..??

One-HyeWon replied:
뭐가 정답인진 모르겠지만 저였다면 SnsContent를 Container 처럼 써서 layout 관련 스타일만 줄 것 같아요!
span에 스타일을 준 스타일드 컴포넌트로 주면 SNS 텍스트에 스타일을 주려는 걸 명확히 알 수 있어서요!

One-HyeWon reviewed (approved):
2주차도 고생 많으셨습니다 나영님!!

간단한 코멘트 달아 두었으니 참고 해주세요~

nabbang6 committed:
fix: Header 스타일 수정

woneeeee commented on 서나영-Blog/src/components/modal/LoginModal.tsx:
```diff
@@ -0,0 +1,271 @@
+  const [password, setPassword] = useState('');
+  const [error, setError] = useState('');
+
+  const validateForm = () => {
+    if (!email) {
+      setError('이메일을 입력해주세요.');
+      return false;
+    } else if (!email.includes('@')) {
+      setError('이메일 형식이 적합하지 않습니다.');
+      return false;
+    } else if (password.length < 6) {
+      // 임시 에러 메시지
+      setError('비밀번호가 일치하지 않습니다.');
+      return false;
+    }
+    setError('');
+    return true;
+  };
```
이런식으로 구현해도 좋은데 저는 
```
import { useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';
import { signupSchema, SignupSchema } from '@/schema/auth';
```
이렇게 각 schema들을 작성해서 해당 라이브러리를 사용했었는데 이런 식으로 해도 깔끔하게 잘 작성하실 수 있을 것 같아용! 이건 저의 개인적인 코드들이라서... 그냥 한 번 코멘트 남겨봤습니당! ㅎㅎ

nabbang6 replied:
이런 라이브러리들이 있군요..!! 처음 보는 라이브러리들이라 인터넷에 내용 더 찾아보았는데... zod 사용해서 검증하는 게 확실히 강점이 더 많아 보여서 회원가입 부분에도 같이 적용해줬습니다! 코멘트 넘 감사드려요 ㅇ///ㅇ !!👍

woneeeee commented on 서나영-Blog/src/components/layout/header/Header.tsx:
```diff
@@ -0,0 +1,83 @@
+import { useState } from 'react';
+import { GITLOG, Reorder } from '@/assets';
+import ChatandMore from './ChatandMore';
+import DelandCreate from './DelandCreate';
+import Button from '@/components/ui/Button';
+import Sidebar from '../Sidebar/Sidebar';
```
절대경로로 수정해주시면 더 좋을 것 가타요!

woneeeee reviewed (approved):
고생하셨습니다! 해야될 것도 많은데 빠르게 구현해주셧네용 👍🏻

yezzan9 commented on 서나영-Blog/src/components/auth/signup/ProfileUpload.tsx:
```diff
@@ -0,0 +1,62 @@
+    <ProfileContainer>
+      <Label>프로필 사진</Label>
+      <ProfileWrapper>
+        <Profile width='90px' height='90px' />
```
현재 Profile이라는 이미지를 그대로 렌더링하고 있는데, 실제로 이미지를 업로드하게 되면 해당 사진이 업로드한 이미지로 변경되어야 하거든요 !
이 부분을 고려하여 동적으로 사용할 수 있게 수정해주시면 나중에 편할 것 같아요!🙌

nabbang6 replied:
넵 반영하였습니다!!

yezzan9 commented on 서나영-Blog/src/components/auth/signup/SignupHeader.tsx:
```diff
@@ -0,0 +1,41 @@
+  align-items: flex-start;
+`;
+
+const SignupHeader = () => {
```
이 부분이 실제 헤더는 아니기 때문에 SignupHeader라고 네이밍하면 헷갈릴 수 있을 것 같아요!
`SignupTitle` `InfoSection`? 네이밍이 정말 어렵긴 한데,, 헤더를 포함하지 않는 단어로 수정하면 좋을 것 같아요🤗

nabbang6 replied:
SignupTitle로 수정하였습니다~! ദ്ദി ｡･･｡)

yezzan9 commented on 서나영-Blog/src/components/modal/LoginModal.tsx:
```diff
@@ -0,0 +1,271 @@
+          <Clear width={'24px'} height={'24px'} />
+        </CloseButton>
+        <LoginLeftSection>
+          <LogoContainer>
+            <LogoWrapper>
+              <GITLOG width={282.692} height={76.029} />
+            </LogoWrapper>
+          </LogoContainer>
```
GITLOG 아이콘에 대해 `LogoContainer`와 `LogoWrapper` 이렇게 이중으로 감싸져 있는데 혹시 이유가 있을까용?-?
자식태그가 하나뿐이니 스타일도 하나의 컴포넌트에 모두 구현할 수 있지 않나 해서요 !

nabbang6 replied:
피그마에 있던 디자인 그대로 가져오려 하다보니 중첩해서 감싸게 되었는데... 생각해보니 한 번만 감싸도 되는 부분이었네요,, ◠//◠  수정하였습니다!

yezzan9 commented on 서나영-Blog/src/components/ui/Modal.tsx:
이번에 구현해주신 부분은 아니지만 이제 봐서,,
모달에 버튼 색상, 버튼 내 텍스트 모두 변경될 수 있는 값이기 때문에 동적으로 사용할 수 있도록 수정해주시면 좋을 것 같아요👀

yezzan9 reviewed (commented):
나영님 2주차 과제도 수고하셨습니다!
몇가지 코멘트 남겨드렸으니 확인 후 나영님 생각도 공유해주세요✨

nabbang6 committed:
feat: react-hook-form 설치

nabbang6 committed:
feat: @hookform/resolvers & zod 설치

nabbang6 committed:
feat: svg 색상을 current로 설정

nabbang6 committed:
refactor: SignupHeader -> SignupTitle로 파일명 변경

nabbang6 committed:
refactor: 절대 경로 적용, fix: GITLOG 로고 표시 오류 수정

nabbang6 committed:
feat: 모달 버튼 속성 동적 할당 기능 추가
- 불필요한 상태 관리 코드 삭제
- description을 선택적 props로 변경
- isOpen ->  으로 변경
- 모달 버튼의 backgroundColor, text 속성을 동적으로 할당할 수 있게 수정

nabbang6 committed:
feat: submit 버튼 타입 추가
- Create 아이콘 색상을 iconFill props로 할당 가능
- onClick 핸들러 추가

nabbang6 committed:
feat: 에러메시지 표시 기능 추가, refactor: SignupInput의 props 수정

nabbang6 committed:
feat: 회원가입 페이지 - 회원가입 버튼 submit 기능 추가
- zod, react-hook-form 사용해서 유효성 검사
- 불필요한 스타일 코드 제거 및 SnsText 스타일 추가

nabbang6 committed:
refactor: isOpen ->  변경

nabbang6 committed:
feat: auth 스키마 추가

nabbang6 committed:
feat: 프로필 이미지 동적 할당 기능 추가

nabbang6 committed:
feat: 회원가입 페이지 - 스키마로 유효성 검사 & 모달 추가

yezzan9 reviewed (approved):
수정해주신 부분 모두 확인했습니당! 수고하셨어요🫶🏻
3주차 과제도 파이팅입니다💪🏻
